utils/wasm-gen/src/config/syscalls/injection.rs

```diff
@@ -0,0 +1,116 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Entities describing possible injection types for each sys-call.
+//!
+//! Types here are used to create [`crate::SysCallsConfig`].
+
+use crate::InvocableSysCall;
+
+use gear_wasm_instrument::syscalls::SysCallName;
+use std::{collections::HashMap, ops::RangeInclusive};
+
+/// Possible injection type.
```

Comment:
I would add more docs/clarifications. For example, variants of `InjectionType` should be decsribed thoroughly as it's the code potentially used by entities of external crates.

Reply:
that's not only about the type, but generally the module.

Reply:
done in d65fe0e3ee3c92bfd27098c1eea86963b79e9697